quartz-core/src/main/java/org/quartz/impl/jdbcjobstore/StdJDBCDelegate.java

```diff
@@ -1158,9 +1158,9 @@ public int updateTrigger(Connection conn, OperableTrigger trigger, String state,
             JobDetail jobDetail) throws SQLException, IOException {
 
         // save some clock cycles by unnecessarily writing job data blob ...
-        boolean updateJobData = trigger.getJobDataMap().isDirty();
+        boolean updateJobData = trigger.getJobDataMap().isDirty() && (trigger.getJobDataMap().size() > 0);
```

Comment:
Regarding the size > 0 check -- what if the size used to be > 0, and now the size is 0 because entries were removed, should not the change (the emptiness) be persisted?